src/components/LibraryItem.tsx

```diff
@@ -248,18 +251,20 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
         }
     }
 
-    onLibraryItemHoveredOn() {
+    onLibraryItemMouseLeave() {
         let libraryContainer = this.props.libraryContainer;
         if (this.props.data.childItems.length == 0) {
-            libraryContainer.raiseEvent(libraryContainer.props.libraryController.ItemHoveredOnEventName,
-                this.props.data.contextData);
+            libraryContainer.raiseEvent(libraryContainer.props.libraryController.ItemMouseLeaveEventName, true);
         }
     }
 
-    onLibraryItemMouseLeave() {
+    onLibraryItemMouseEnter() {
         let libraryContainer = this.props.libraryContainer;
         if (this.props.data.childItems.length == 0) {
-            libraryContainer.raiseEvent(libraryContainer.props.libraryController.ItemMouseLeaveEventName, true);
+            var rect = ReactDOM.findDOMNode(this).getBoundingClientRect();
+            let middle = (rect.bottom + rect.top)/2.0;
```

Comment:
should we align our tooltip with the top position of the item or the middle? To me, top position makes better sense.